From b9084fab02c009a593b604562a69f36a5915c8e5 Mon Sep 17 00:00:00 2001
From: Denis Laxalde <denis@laxalde.org>
Date: Sat, 2 Mar 2024 07:40:36 +0100
Subject: [PATCH] Remove pytest.deprecated_call() in TestAssoc::test_unknown
 (#1249)

assoc() no longer raises a deprecation warning since commit
22ae8473fb88d6e585b05c709e81e1a46398a649 but the 'with
pytest.deprecated_call():' in that test was not removed then (in
contrast with other test cases).

Maybe this got unnoticed due to a pytest bug?
In any case, using pytest 8+ (and keeping deprecated_call()) shows that
no warning is raised and the test fails.

Removing the upper bound on pytest in dev dependencies as tests now
pass with pytest 8.0.

Fix #1233.

Co-authored-by: Hynek Schlawack <hs@ox.cx>
---
 pyproject.toml      | 3 +--
 tests/test_funcs.py | 4 +---
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/tests/test_funcs.py b/tests/test_funcs.py
index 044aaab..398ba35 100644
--- a/tests/test_funcs.py
+++ b/tests/test_funcs.py
@@ -600,9 +600,7 @@ class TestAssoc:
         AttrsAttributeNotFoundError.
         """
         # No generated class will have a four letter attribute.
-        with pytest.raises(
-            AttrsAttributeNotFoundError
-        ) as e, pytest.deprecated_call():
+        with pytest.raises(AttrsAttributeNotFoundError) as e:
             assoc(C(), aaaa=2)
 
         assert (f"aaaa is not an attrs attribute on {C!r}.",) == e.value.args
-- 
2.44.0

